mentat/code_file_manager.py

```diff
@@ -93,7 +94,86 @@ def _abs_file_paths_from_list(paths: Iterable[str], check_for_text: bool = True)
                     nonignored_files,
                 )
             )
-    return file_paths_direct, file_paths_from_dirs
+
+    files_from_dirs = [File(path) for path in file_paths_from_dirs]
+    return files_direct, files_from_dirs
+
+
+def _abs_file_paths_from_list(paths: Iterable[str], check_for_text: bool = True):
+    files_direct, files_from_dirs = _abs_files_from_list(paths, check_for_text)
+    return set(map(lambda f: f.path, files_direct)), set(
+        map(lambda f: f.path, files_from_dirs)
+    )
+
+
+class Interval:
+    def __init__(
+        self,
+        start: int,
+        end: int,
+    ):
+        self.start = start
+        self.end = end
+
+    def contains(self, k):
+        return self.start <= k <= self.end
+
+
+def parse_intervals(interval_string: str) -> list[Interval] | None:
+    try:
+        intervals = []
+        for interval in interval_string.split(","):
+            interval = interval.split("-", 1)
+            if len(interval) == 1:
+                intervals += [Interval(int(interval[0]), int(interval[0]))]
+            else:
+                intervals += [Interval(int(interval[0]), int(interval[1]))]
+        return intervals
+    except (ValueError, IndexError):
+        return None
+
+
+class File:
```

Comment:
Hmm, I definitely agree that naming it File is a bit weird; maybe naming it CodeFile would be better? Not sure how much I like that either, but I think it's better.

Reply:
I also think we should refactor File and Interval into another file (codefile.py?)